temoa/extensions/get_comm_tech.py

```
@@ -1,22 +1,24 @@
 import sys
 from collections import OrderedDict
+from typing import Any
 
 
-def get_tperiods(inp_f):
+def get_tperiods(inp_f: str) -> dict[str, list[int]]:
     file_ty = re.search(r'(\w+)\.(\w+)\b', inp_f)  # Extract the input filename and extension
 
     if not file_ty:
-        raise 'The file type %s is not recognized.' % inp_f
+        raise Exception(f'The file type {inp_f} is not recognized.')
```

Ruff (0.14.10)

Create your own exception

(TRY002)